### DISCUSSION TITLE
Coordinate system error

### DISCUSSION THREAD

**Discussion by Phome95** [r-lidar/lidR#585] [General]
Hello, I am attempting to use the silva2016 package:

```r
chm <- raster("chm.tif")
ttops <- locate_trees(chm, lmf(3))
crowns <- silva2016(chm, ttops)()
```
however this error comes up:

```
Error in geos_op2_geom("intersection", x, y, ...) : 
  st_crs(x) == st_crs(y) is not TRUE
```

However both coordinate systems are the same.

![image](https://user-images.githubusercontent.com/100789966/169665614-19800b62-93f4-4d98-aaeb-a7c323d6a732.png)

**Comment by Phome95** [comment]
This is my script, thank you for the help!:

```r
library(lidR)
library(treetop)
library(raster)
library(rgdal)
# Check memory limit
memory.limit()
# Change memory limit
memory.limit(size = 56000)
chm_disk<- raster("C:/Users/ezxph2/OneDrive - The University of Nottingham/MRes/dissertation/Data/LiDAR/chm_clean2.tif")
chm_memory <- readAll(chm_disk) #put chm onto memory
ttops <- locate_trees(chm_memory, lmf(3))

st_crs(ttops)#identify crs
st_crs(chm_memory)
ttops <- st_transform(ttops, st_crs(chm_memory)) #ensure coordinate reference system are the same
crowns <- silva2016(chm_memory, ttops)()
```

**Reply by Jean-Romain** [reply]
Please share data to make your example reproductible

**Reply by Phome95** [reply]
https://drive.google.com/file/d/1pzQ3crlBwp3LS-vvqoPef8V_io1W-Iq5/view?usp=sharing

Unfortunately I had some issues sharing the chm to github as the file is around 200mb, but this link to my google drive should work

**Comment by Jean-Romain** [comment]
#586
